src/errorpage.cc

```diff
@@ -841,6 +841,7 @@ ErrorState::~ErrorState()
     wordlistDestroy(&ftp.server_msg);
     safe_free(ftp.request);
     safe_free(ftp.reply);
+    safe_free(ftp.cwd_msg);
```

Comment:
Please also free `cwd_msg` just before assigning it in Ftp::Gateway::completedListing(). Doing so may not fix any current bugs, but it is the right thing to do in a PR managing `cwd_msg` lifetime.